### ISSUE TITLE
uploading images in local-server throws `Aws\S3\Exception\S3Exception: Error executing "ListObjects" on`

### ISSUE THREAD

**Issue by owaincuvelier** [humanmade/altis-local-server#542]
Steps to reproduce:
1. Create new fresh Altis v12 install locally.
2. Upload any image
3. Generic error on front end and PHP error in logs can be found
```
 [15-Nov-2022 12:08:14 UTC] PHP Fatal error: Uncaught exception 'Aws\S3\Exception\S3Exception' with message 'Error executing "ListObjects" on "https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploadswp-personal-data-exports&encoding-type=url"; AWS HTTP error: Client error: `GET https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploadswp-personal-data-exports&encoding-type=url` resulted in a `400 Bad Request` response:
matt-test-php | 
matt-test-php | XMinioInvalidObjectNameObject name contains unsuppor (truncated...)
matt-test-php | XMinioInvalidObjectName (client): Object name contains unsupported characters. - 
matt-test-php | XMinioInvalidObjectNameObject name contains unsupported characters.s3-matt-test/s3-matt-test//us-east-11727C0BAAB6303A647004af2-6303-4eae-b928-7b165e19285f'
matt-test-php | 
matt-test-php | GuzzleHttp\Exception\ClientException: Client error: `GET https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploadswp-personal-data-exports&encoding-type=url` resulted in a `400 Bad Request` response:
matt-test-php | 
matt-test-php | XMinioInvalidObjectNameObject name contains unsuppor (truncated...)
matt-test-php | in /usr/src/app/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113
matt-test-php | Stack trace:
matt-test-php | #0 /usr/src/app/vendor/guzzlehttp/guzzle/src/Middleware.php(69): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL)
matt-test-php | humanmade/altis-cloud#1 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(204): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response))
matt-test-php | humanmade/altis-cloud#2 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(153): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL)
matt-test-php | humanmade/altis-cloud#3 /usr/src/app/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
matt-test-php | humanmade/altis-cloud#4 /usr/src/app/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php(159): GuzzleHttp\Promise\TaskQueue->run()
matt-test-php | humanmade/altis-cloud#5 /usr/src/app/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php(184): GuzzleHttp\Handler\CurlMultiHandler->tick()
matt-test-php | humanmade/altis-cloud#6 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Handler\CurlMultiHandler->execute(true)
matt-test-php | humanmade/altis-cloud#7 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn()
matt-test-php | humanmade/altis-cloud#8 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#9 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
matt-test-php | humanmade/altis-cloud#10 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#11 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
matt-test-php | humanmade/altis-cloud#12 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#13 /usr/src/app/vendor/aws/aws-sdk-php/src/AwsClientTrait.php(58): GuzzleHttp\Promise\Promise->wait()
matt-test-php | humanmade/altis-cloud#14 /usr/src/app/vendor/aws/aws-sdk-php/src/ResultPaginator.php(131): Aws\AwsClient->execute(Object(Aws\Command))
matt-test-php | humanmade/altis-cloud#15 /usr/src/app/vendor/aws/aws-sdk-php/src/functions.php(52): Aws\ResultPaginator->valid()
matt-test-php | humanmade/altis-cloud#16 /usr/src/app/vendor/aws/aws-sdk-php/src/functions.php(69): Aws\map(Object(Aws\ResultPaginator), Object(Closure))
matt-test-php | humanmade/altis-cloud#17 [internal function]: Aws\flatmap(Object(Aws\ResultPaginator), Object(Closure))
matt-test-php | humanmade/altis-cloud#18 /usr/src/app/vendor/humanmade/s3-uploads/inc/class-stream-wrapper.php(695): Generator->valid()
matt-test-php | humanmade/altis-cloud#19 [internal function]: S3_Uploads\Stream_Wrapper->dir_readdir()
matt-test-php | humanmade/altis-cloud#20 /usr/src/app/wordpress/wp-admin/includes/file.php(151): readdir(Resource id #1733)
matt-test-php | humanmade/altis-cloud#21 /usr/src/app/wordpress/wp-includes/functions.php(7929): list_files('s3://s3-matt-te...', 100, Array)
matt-test-php | humanmade/altis-cloud#22 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(307): wp_privacy_delete_old_export_files()
matt-test-php | humanmade/altis-cloud#23 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters('', Array)
matt-test-php | humanmade/altis-cloud#24 /usr/src/app/wordpress/wp-includes/plugin.php(524): WP_Hook->do_action(Array)
matt-test-php | humanmade/altis-cloud#25 /usr/src/app/vendor/humanmade/cavalcade/inc/class-command.php(44): do_action_ref_array('wp_privacy_dele...', Array)
matt-test-php | humanmade/altis-cloud#26 [internal function]: HM\Cavalcade\Plugin\Command->run(Array, Array)
matt-test-php | humanmade/altis-cloud#27 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php(100): call_user_func(Array, Array, Array)
matt-test-php | humanmade/altis-cloud#28 [internal function]: WP_CLI\Dispatcher\CommandFactory::WP_CLI\Dispatcher\{closure}(Array, Array)
matt-test-php | humanmade/altis-cloud#29 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(491): call_user_func(Object(Closure), Array, Array)
matt-test-php | humanmade/altis-cloud#30 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(417): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
matt-test-php | humanmade/altis-cloud#31 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(440): WP_CLI\Runner->run_command(Array, Array)
matt-test-php | humanmade/altis-cloud#32 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1237): WP_CLI\Runner->run_command_and_exit()
matt-test-php | humanmade/altis-cloud#33 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/LaunchRunner.php(28): WP_CLI\Runner->start()
matt-test-php | humanmade/altis-cloud#34 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/bootstrap.php(78): WP_CLI\Bootstrap\LaunchRunner->process(Object(WP_CLI\Bootstrap\BootstrapState))
matt-test-php | humanmade/altis-cloud#35 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/wp-cli.php(27): WP_CLI\bootstrap()
matt-test-php | humanmade/altis-cloud#36 phar:///usr/local/bin/wp/php/boot-phar.php(11): include('phar:///usr/loc...')
matt-test-php | humanmade/altis-cloud#37 /usr/local/bin/wp(4): include('phar:///usr/loc...')
matt-test-php | humanmade/altis-cloud#38 {main}
matt-test-php | 
matt-test-php | Next Aws\S3\Exception\S3Exception: Error executing "ListObjects" on "https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploadswp-personal-data-exports&encoding-type=url"; AWS HTTP error: Client error: `GET https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploadswp-personal-data-exports&encoding-type=url` resulted in a `400 Bad Request` response:
matt-test-php | 
matt-test-php | XMinioInvalidObjectNameObject name contains unsuppor (truncated...)
matt-test-php | XMinioInvalidObjectName (client): Object name contains unsupported characters. - 
matt-test-php | XMinioInvalidObjectNameObject name contains unsupported characters.s3-matt-test/s3-matt-test//us-east-11727C0BAAB6303A647004af2-6303-4eae-b928-7b165e19285f in /usr/src/app/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php:195
matt-test-php | Stack trace:
matt-test-php | #0 /usr/src/app/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php(97): Aws\WrappedHttpHandler->parseError(Array, Object(GuzzleHttp\Psr7\Request), Object(Aws\Command), Array)
matt-test-php | humanmade/altis-cloud#1 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(204): Aws\WrappedHttpHandler->Aws\{closure}(Array)
matt-test-php | humanmade/altis-cloud#2 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(169): GuzzleHttp\Promise\Promise::callHandler(2, Array, NULL)
matt-test-php | humanmade/altis-cloud#3 /usr/src/app/vendor/guzzlehttp/promises/src/RejectedPromise.php(42): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}(Array)
matt-test-php | humanmade/altis-cloud#4 /usr/src/app/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\RejectedPromise::GuzzleHttp\Promise\{closure}()
matt-test-php | humanmade/altis-cloud#5 /usr/src/app/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php(159): GuzzleHttp\Promise\TaskQueue->run()
matt-test-php | humanmade/altis-cloud#6 /usr/src/app/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php(184): GuzzleHttp\Handler\CurlMultiHandler->tick()
matt-test-php | humanmade/altis-cloud#7 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Handler\CurlMultiHandler->execute(true)
matt-test-php | humanmade/altis-cloud#8 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn()
matt-test-php | humanmade/altis-cloud#9 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#10 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
matt-test-php | humanmade/altis-cloud#11 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#12 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
matt-test-php | humanmade/altis-cloud#13 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#14 /usr/src/app/vendor/aws/aws-sdk-php/src/AwsClientTrait.php(58): GuzzleHttp\Promise\Promise->wait()
matt-test-php | humanmade/altis-cloud#15 /usr/src/app/vendor/aws/aws-sdk-php/src/ResultPaginator.php(131): Aws\AwsClient->execute(Object(Aws\Command))
matt-test-php | humanmade/altis-cloud#16 /usr/src/app/vendor/aws/aws-sdk-php/src/functions.php(52): Aws\ResultPaginator->valid()
matt-test-php | humanmade/altis-cloud#17 /usr/src/app/vendor/aws/aws-sdk-php/src/functions.php(69): Aws\map(Object(Aws\ResultPaginator), Object(Closure))
matt-test-php | humanmade/altis-cloud#18 [internal function]: Aws\flatmap(Object(Aws\ResultPaginator), Object(Closure))
matt-test-php | humanmade/altis-cloud#19 /usr/src/app/vendor/humanmade/s3-uploads/inc/class-stream-wrapper.php(695): Generator->valid()
matt-test-php | humanmade/altis-cloud#20 [internal function]: S3_Uploads\Stream_Wrapper->dir_readdir()
matt-test-php | humanmade/altis-cloud#21 /usr/src/app/wordpress/wp-admin/includes/file.php(151): readdir(Resource id #1733)
matt-test-php | humanmade/altis-cloud#22 /usr/src/app/wordpress/wp-includes/functions.php(7929): list_files('s3://s3-matt-te...', 100, Array)
matt-test-php | humanmade/altis-cloud#23 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(307): wp_privacy_delete_old_export_files()
matt-test-php | humanmade/altis-cloud#24 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(331): WP_Hook->apply_filters('', Array)
matt-test-php | humanmade/altis-cloud#25 /usr/src/app/wordpress/wp-includes/plugin.php(524): WP_Hook->do_action(Array)
matt-test-php | humanmade/altis-cloud#26 /usr/src/app/vendor/humanmade/cavalcade/inc/class-command.php(44): do_action_ref_array('wp_privacy_dele...', Array)
matt-test-php | humanmade/altis-cloud#27 [internal function]: HM\Cavalcade\Plugin\Command->run(Array, Array)
matt-test-php | humanmade/altis-cloud#28 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/CommandFactory.php(100): call_user_func(Array, Array, Array)
matt-test-php | humanmade/altis-cloud#29 [internal function]: WP_CLI\Dispatcher\CommandFactory::WP_CLI\Dispatcher\{closure}(Array, Array)
matt-test-php | humanmade/altis-cloud#30 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Dispatcher/Subcommand.php(491): call_user_func(Object(Closure), Array, Array)
matt-test-php | humanmade/altis-cloud#31 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(417): WP_CLI\Dispatcher\Subcommand->invoke(Array, Array, Array)
matt-test-php | humanmade/altis-cloud#32 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(440): WP_CLI\Runner->run_command(Array, Array)
matt-test-php | humanmade/altis-cloud#33 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Runner.php(1237): WP_CLI\Runner->run_command_and_exit()
matt-test-php | humanmade/altis-cloud#34 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/WP_CLI/Bootstrap/LaunchRunner.php(28): WP_CLI\Runner->start()
matt-test-php | humanmade/altis-cloud#35 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/bootstrap.php(78): WP_CLI\Bootstrap\LaunchRunner->process(Object(WP_CLI\Bootstrap\BootstrapState))
matt-test-php | humanmade/altis-cloud#36 phar:///usr/local/bin/wp/vendor/wp-cli/wp-cli/php/wp-cli.php(27): WP_CLI\bootstrap()
matt-test-php | humanmade/altis-cloud#37 phar:///usr/local/bin/wp/php/boot-phar.php(11): include('phar:///usr/loc...')
matt-test-php | humanmade/altis-cloud#38 /usr/local/bin/wp(4): include('phar:///usr/loc...')
matt-test-php | humanmade/altis-cloud#39 {main}
matt-test-php | thrown in /usr/src/app/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php on line 195
matt-test-php | [15-Nov-2022 12:08:15 UTC] PHP Warning: Undefined array key "file" in /usr/src/app/vendor/johnbillion/query-monitor/collectors/php_errors.php on line 380
matt-test-php | [15-Nov-2022 12:08:15 UTC] PHP Warning: Undefined array key "line" in /usr/src/app/vendor/johnbillion/query-monitor/collectors/php_errors.php on line 380
matt-test-php | [15-Nov-2022 12:08:15 UTC] PHP Warning: Undefined array key "file" in /usr/src/app/vendor/johnbillion/query-monitor/collectors/php_errors.php on line 380
matt-test-php | [15-Nov-2022 12:08:15 UTC] PHP Warning: Undefined array key "line" in /usr/src/app/vendor/johnbillion/query-monitor/collectors/php_errors.php on line 380
matt-test-php | [15-Nov-2022 12:08:15 UTC] PHP Fatal error: Uncaught exception 'Aws\S3\Exception\S3Exception' with message 'Error executing "ListObjects" on "https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploads202211Felis_silvestris_silvestris_small_gradual_decrease_of_quality&encoding-type=url"; AWS HTTP error: Client error: `GET https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploads202211Felis_silvestris_silvestris_small_gradual_decrease_of_quality&encoding-type=url` resulted in a `400 Bad Request` response:
matt-test-php | 
matt-test-php | XMinioInvalidObjectNameObject name contains unsuppor (truncated...)
matt-test-php | XMinioInvalidObjectName (client): Object name contains unsupported characters. - 
matt-test-php | XMinioInvalidObjectNameObject name contains unsupported characters.s3-matt-test/s3-matt-test//us-east-11727C0BADC1BC55E47004af2-6303-4eae-b928-7b165e19285f'
matt-test-php | 
matt-test-php | GuzzleHttp\Exception\ClientException: Client error: `GET https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploads202211Felis_silvestris_silvestris_small_gradual_decrease_of_quality&encoding-type=url` resulted in a `400 Bad Request` response:
matt-test-php | 
matt-test-php | XMinioInvalidObjectNameObject name contains unsuppor (truncated...)
matt-test-php | in /usr/src/app/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php:113
matt-test-php | Stack trace:
matt-test-php | #0 /usr/src/app/vendor/guzzlehttp/guzzle/src/Middleware.php(69): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL)
matt-test-php | humanmade/altis-cloud#1 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(204): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response))
matt-test-php | humanmade/altis-cloud#2 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(153): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL)
matt-test-php | humanmade/altis-cloud#3 /usr/src/app/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
matt-test-php | humanmade/altis-cloud#4 /usr/src/app/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php(159): GuzzleHttp\Promise\TaskQueue->run()
matt-test-php | humanmade/altis-cloud#5 /usr/src/app/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php(184): GuzzleHttp\Handler\CurlMultiHandler->tick()
matt-test-php | humanmade/altis-cloud#6 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Handler\CurlMultiHandler->execute(true)
matt-test-php | humanmade/altis-cloud#7 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn()
matt-test-php | humanmade/altis-cloud#8 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#9 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
matt-test-php | humanmade/altis-cloud#10 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#11 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
matt-test-php | humanmade/altis-cloud#12 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#13 /usr/src/app/vendor/aws/aws-sdk-php/src/AwsClientTrait.php(58): GuzzleHttp\Promise\Promise->wait()
matt-test-php | humanmade/altis-cloud#14 /usr/src/app/vendor/aws/aws-sdk-php/src/ResultPaginator.php(131): Aws\AwsClient->execute(Object(Aws\Command))
matt-test-php | humanmade/altis-cloud#15 /usr/src/app/vendor/aws/aws-sdk-php/src/functions.php(52): Aws\ResultPaginator->valid()
matt-test-php | humanmade/altis-cloud#16 /usr/src/app/vendor/aws/aws-sdk-php/src/functions.php(69): Aws\map(Object(Aws\ResultPaginator), Object(Closure))
matt-test-php | humanmade/altis-cloud#17 [internal function]: Aws\flatmap(Object(Aws\ResultPaginator), Object(Closure))
matt-test-php | humanmade/altis-cloud#18 /usr/src/app/vendor/humanmade/s3-uploads/inc/class-stream-wrapper.php(695): Generator->valid()
matt-test-php | humanmade/altis-cloud#19 [internal function]: S3_Uploads\Stream_Wrapper->dir_readdir()
matt-test-php | humanmade/altis-cloud#20 /usr/src/app/vendor/humanmade/s3-uploads/inc/class-plugin.php(668): scandir('s3://s3-matt-te...')
matt-test-php | humanmade/altis-cloud#21 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(307): S3_Uploads\Plugin->get_files_for_unique_filename_file_list(NULL, 's3://s3-matt-te...', 'Felis_silvestri...')
matt-test-php | humanmade/altis-cloud#22 /usr/src/app/wordpress/wp-includes/plugin.php(191): WP_Hook->apply_filters(NULL, Array)
matt-test-php | humanmade/altis-cloud#23 /usr/src/app/wordpress/wp-includes/functions.php(2642): apply_filters('pre_wp_unique_f...', NULL, 's3://s3-matt-te...', 'Felis_silvestri...')
matt-test-php | humanmade/altis-cloud#24 /usr/src/app/wordpress/wp-admin/includes/file.php(953): wp_unique_filename('s3://s3-matt-te...', 'Felis_silvestri...', NULL)
matt-test-php | humanmade/altis-cloud#25 /usr/src/app/wordpress/wp-admin/includes/file.php(1074): _wp_handle_upload(Array, Array, '2022-11-15 12:0...', 'wp_handle_uploa...')
matt-test-php | humanmade/altis-cloud#26 /usr/src/app/wordpress/wp-admin/includes/media.php(303): wp_handle_upload(Array, Array, '2022-11-15 12:0...')
matt-test-php | humanmade/altis-cloud#27 /usr/src/app/wordpress/wp-admin/async-upload.php(114): media_handle_upload('async-upload', 0)
matt-test-php | humanmade/altis-cloud#28 {main}
matt-test-php | 
matt-test-php | Next Aws\S3\Exception\S3Exception: Error executing "ListObjects" on "https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploads202211Felis_silvestris_silvestris_small_gradual_decrease_of_quality&encoding-type=url"; AWS HTTP error: Client error: `GET https://s3-matt-test.altis.dev/s3-matt-test//?delimiter=&prefix=uploads202211Felis_silvestris_silvestris_small_gradual_decrease_of_quality&encoding-type=url` resulted in a `400 Bad Request` response:
matt-test-php | 
matt-test-php | XMinioInvalidObjectNameObject name contains unsuppor (truncated...)
matt-test-php | XMinioInvalidObjectName (client): Object name contains unsupported characters. - 
matt-test-php | XMinioInvalidObjectNameObject name contains unsupported characters.s3-matt-test/s3-matt-test//us-east-11727C0BADC1BC55E47004af2-6303-4eae-b928-7b165e19285f in /usr/src/app/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php:195
matt-test-php | Stack trace:
matt-test-php | #0 /usr/src/app/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php(97): Aws\WrappedHttpHandler->parseError(Array, Object(GuzzleHttp\Psr7\Request), Object(Aws\Command), Array)
matt-test-php | humanmade/altis-cloud#1 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(204): Aws\WrappedHttpHandler->Aws\{closure}(Array)
matt-test-php | humanmade/altis-cloud#2 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(169): GuzzleHttp\Promise\Promise::callHandler(2, Array, NULL)
matt-test-php | humanmade/altis-cloud#3 /usr/src/app/vendor/guzzlehttp/promises/src/RejectedPromise.php(42): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}(Array)
matt-test-php | humanmade/altis-cloud#4 /usr/src/app/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\RejectedPromise::GuzzleHttp\Promise\{closure}()
matt-test-php | humanmade/altis-cloud#5 /usr/src/app/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php(159): GuzzleHttp\Promise\TaskQueue->run()
matt-test-php | humanmade/altis-cloud#6 /usr/src/app/vendor/guzzlehttp/guzzle/src/Handler/CurlMultiHandler.php(184): GuzzleHttp\Handler\CurlMultiHandler->tick()
matt-test-php | humanmade/altis-cloud#7 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Handler\CurlMultiHandler->execute(true)
matt-test-php | humanmade/altis-cloud#8 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn()
matt-test-php | humanmade/altis-cloud#9 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#10 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
matt-test-php | humanmade/altis-cloud#11 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#12 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
matt-test-php | humanmade/altis-cloud#13 /usr/src/app/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending()
matt-test-php | humanmade/altis-cloud#14 /usr/src/app/vendor/aws/aws-sdk-php/src/AwsClientTrait.php(58): GuzzleHttp\Promise\Promise->wait()
matt-test-php | humanmade/altis-cloud#15 /usr/src/app/vendor/aws/aws-sdk-php/src/ResultPaginator.php(131): Aws\AwsClient->execute(Object(Aws\Command))
matt-test-php | humanmade/altis-cloud#16 /usr/src/app/vendor/aws/aws-sdk-php/src/functions.php(52): Aws\ResultPaginator->valid()
matt-test-php | humanmade/altis-cloud#17 /usr/src/app/vendor/aws/aws-sdk-php/src/functions.php(69): Aws\map(Object(Aws\ResultPaginator), Object(Closure))
matt-test-php | humanmade/altis-cloud#18 [internal function]: Aws\flatmap(Object(Aws\ResultPaginator), Object(Closure))
matt-test-php | humanmade/altis-cloud#19 /usr/src/app/vendor/humanmade/s3-uploads/inc/class-stream-wrapper.php(695): Generator->valid()
matt-test-php | humanmade/altis-cloud#20 [internal function]: S3_Uploads\Stream_Wrapper->dir_readdir()
matt-test-php | humanmade/altis-cloud#21 /usr/src/app/vendor/humanmade/s3-uploads/inc/class-plugin.php(668): scandir('s3://s3-matt-te...')
matt-test-php | humanmade/altis-cloud#22 /usr/src/app/wordpress/wp-includes/class-wp-hook.php(307): S3_Uploads\Plugin->get_files_for_unique_filename_file_list(NULL, 's3://s3-matt-te...', 'Felis_silvestri...')
matt-test-php | humanmade/altis-cloud#23 /usr/src/app/wordpress/wp-includes/plugin.php(191): WP_Hook->apply_filters(NULL, Array)
matt-test-php | humanmade/altis-cloud#24 /usr/src/app/wordpress/wp-includes/functions.php(2642): apply_filters('pre_wp_unique_f...', NULL, 's3://s3-matt-te...', 'Felis_silvestri...')
matt-test-php | humanmade/altis-cloud#25 /usr/src/app/wordpress/wp-admin/includes/file.php(953): wp_unique_filename('s3://s3-matt-te...', 'Felis_silvestri...', NULL)
matt-test-php | humanmade/altis-cloud#26 /usr/src/app/wordpress/wp-admin/includes/file.php(1074): _wp_handle_upload(Array, Array, '2022-11-15 12:0...', 'wp_handle_uploa...')
matt-test-php | humanmade/altis-cloud#27 /usr/src/app/wordpress/wp-admin/includes/media.php(303): wp_handle_upload(Array, Array, '2022-11-15 12:0...')
matt-test-php | humanmade/altis-cloud#28 /usr/src/app/wordpress/wp-admin/async-upload.php(114): media_handle_upload('async-upload', 0)
matt-test-php | humanmade/altis-cloud#29 {main}
matt-test-php | thrown in /usr/src/app/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php on line 195

```

**Comment by kovshenin** [comment]
Spent some more time with @rmccue and @joehoyle debugging this further, here's a brain dump:

* AWS PHP SDK version used: 3.245.0
* Test script to reproduce errors outside of the S3 Uploads and Altis context: https://gist.github.com/kovshenin/9524cc36e09b34d9c56549f6f2ef19e0
* The trailing slash in the endpoint is, otherwise the AWS SDK will strip the bucket name from the path
* ListObjects with a trailing slash (resulting in a double slash) fails with `XMinioInvalidObjectName`
* ListObjectsV2 without a trailing slash does not throw an exception, but does not find any items in the bucket
* ListObjects without a trailing slash + PartitionEndpointProvider does not throw an exception, but does not find any items

These succeed:

* ListObjectsV2 with a trailing slash succeeds and finds items in the bucket
* ListObjects with a trailing slash + PartitionEndpointProvider succeeds and finds correct items

Todo:

* Create a PR for S3 Uploads to replace `ListObjects` with `ListObjectsV2` (the big difference is in pagination)
* Perhaps open an issue on the AWS SDK repo, similar to https://github.com/aws/aws-sdk-php/issues/2563 and demonstrate how `ListObjects` fails with the new `EndpointProviderV2` but `ListObjectsV2` succeeds.

**Comment by kovshenin** [comment]
Opened:

* https://github.com/aws/aws-sdk-php/issues/2575
* https://github.com/humanmade/S3-Uploads/pull/609

**Comment by kovshenin** [comment]
This was fixed upstream and released in https://github.com/aws/aws-sdk-php/releases/tag/3.247.1

**Comment by mikelittle** [comment]
Looks like this can be closed per the comments above.

**Comment by kovshenin** [comment]
Published S3 Uploads 3.0.5 release here: https://github.com/humanmade/S3-Uploads/releases/tag/3.0.5

The altis/cloud module (with a `^3.0.4` constraint) picks the new version up during a `composer update`, tested up until `v10-branch`.
